hibernate-core/src/main/java/org/hibernate/resource/jdbc/ResourceRegistry.java

```diff
@@ -36,6 +36,23 @@
 	 */
 	void register(Statement statement, boolean cancelable);
 
+	/**
+	 * Set the provided SQL {@link String} value to the given {@link Statement} that was registered previously
+	 *
+	 * @param statement JDBC Statement
+	 * @param sql SQL {@link String} value
+	 */
+	void setStatementSql(Statement statement, String sql);
+
+	/**
+	 * Get the SQL {@link String} value associated with the {@link Statement}, if any
+	 *
+	 * @param statement JDBC Statement
+	 *
+	 * @return SQL {@link String} value
```

Comment:
Do we really need the `{@link String}` above? Seems it is so basic to deserve one.

Reply:
True. I could simply replace it with a `string` instead.